iModelCore/ECDb/ECDb/ECSql/ECSqlPragmas.cpp

```diff
@@ -729,5 +729,63 @@ DbResult PragmaIntegrityCheck::Write(PragmaManager::RowSet& rowSet, ECDbCR ecdb,
 }
 
 
+//=======================================================================================
+// PurgeOrphanedRelationships
+//=======================================================================================
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+DbResult PragmaPurgeOrphanRelationships::Read(PragmaManager::RowSet& rowSet, ECDbCR ecdb, const PragmaVal& val, PragmaManager::OptionsMap const& options)
+	{
+	if (!isExperimentalFeatureAllowed(ecdb, options))
+		{
+		ecdb.GetImpl().Issues().ReportV(IssueSeverity::Error, IssueCategory::BusinessProperties, IssueType::ECSQL, ECDbIssueId::ECDb_0729, "'PRAGMA %s' is an experimental feature and is disabled by default.", GetName().c_str());
+		return BE_SQLITE_ERROR;
+		}
+
+	rowSet = std::make_unique<StaticPragmaResult>(ecdb);
+	rowSet->FreezeSchemaChanges();
+
+	bmap<Utf8CP, bvector<Utf8String>> orphanRelationshipRows;
```

Comment:
Why keep the instance IDs as strings? We use them in an IN(), and we don't know how many there may be. I think a virtual set would be the perfect fit for this (BindVirtualSet())